test/codeu/chat/server/BasicControllerTest.java

```diff
@@ -89,4 +91,39 @@ public void testAddMessage() {
         "Check that the message has a valid reference",
         message == null);
   }
+
+  @Test
+  public void testAllowsNormalChar() {
+
+    final String alphaNumeric = "Ab0";
+    final String specialChars = "!~:.@#$%^&*()[]{}?/-=_+|<>";
+
+    assertTrue(
+            "Check that alphanumeric characters are allowed",
+            ClientUser.isValidInput(alphaNumeric)
+    );
+
+    assertTrue(
+            "Check that special characters other than ,;' and whitespace are allowed",
+            ClientUser.isValidInput(specialChars)
+    );
+  }
+
+  @Test
+  public void testPreventsBadChar() {
+
+    final String sqlInjection = ";,'";
+    final String whiteSpace = "   \n\\u000D\\u000C";
```

Comment:
What about empty strings? Or other unicode characters? Are semicolon, comma, apostrophe, and whitespace the only types of characters we're disallowing?